[RHACS] [Docs] [4.7] ROX-30114: Release notes for 4.7.5 (#96076)
openshift/openshift-docs: release_notes/47-release-notes.adoc

* zlib: Out-of-bounds pointer arithmetic in `inftrees.c` link:https://access.redhat.com/security/cve/CVE-2016-9840[(CVE-2016-9840)]
* krb5: Kerberos RC4-HMAC-MD5 checksum vulnerability link:https://access.redhat.com/security/cve/CVE-2025-3576[(CVE-2025-3576)]

[id="about-release-475_{context}"]
== About release version 4.7.5

*Release date*: {ga-date-475}

This release of {product-title-short} 4.7 provides the following bug fixes:

//ROX-29320

> * Before this update, Scanner V4 falsely claimed not to support RHEL 10, although this was always the case. Scanner V4 now correctly indicates that it supports RHEL 10.

Reviewer: > Before this update, Scanner V4 falsely claimed not to support RHEL 10, although this was always the case.

I feel like this may be misinterpreted to mean it was always the case it never supported RHEL 10. Can we change it to something like:

Before this update, Scanner V4 falsely claimed to not support RHEL 10, although it always supported RHEL 10.

> * Before this update, incorrect interpretation of Red Hat Enterprise Linux (RHEL) 10 Common Platform Enumeration (CPE) strings caused Scanner V4 to fail distribution checks on RHEL 10 systems. With this update, an updated RHEL CPE major version pattern resolves the issue, and Scanner V4 can now correctly support RHEL 10.

Reviewer: I think we can simplify this a bit to remove some of the details (also I think it's a bit wrong as it's currently written):

How about:

> Before this update, Scanner V4 incorrectly claimed it did not support RHEL 10 even though it always did. Scanner V4 now correctly indicates that it does support RHEL 10.

//ROX-29755
* Before this update, the failure of Sensor to call stream.Recv() caused gRPC flow control to block image reprocessing every 4 hours. With this update, the reprocessing loop includes a timeout for sending messages to Sensors, which resolves the issue and resumes the image reprocessing as expected.

//ROX-29649
* Before this update, you could observe excessive logging of telemetry collection status, resulting in log spam. With this update, the telemetry collection has been configured to not emit repeated logs continuously, which resolves the issue and significantly reduces the log volume.

//ROX-29716

> * Before this update, image digest mismatches caused signature verification to fail, even when the signatures were valid. With this update, a check verifies image references against signatures, which resolves the issue and allows {ocp} to successfully verify image signatures for secure deployment.

Reviewer 2: > Before this update, image digest mismatches caused signature verification to fail

The fix was related to image digests, however it is expected that a signature with a digest mismatch fails verification; let's see how we can reword this :). Here's an explanation for context, in case it helps choosing the right words:

The problem surfaced when verifying signatures that contain the full image reference, including digest, in the "docker-reference" field. Normally this field only contains the registry and repository, and our algorithm didn't handle the variant that contains the digest as well. This caused valid signatures with the aforementioned format to be reported as invalid.

I hope that is somehow clear, please let me know otherwise. With that in mind, how about something along the lines of:

"Before this update, valid signatures with a particular payload format may be reported as invalid. With this update, the signature verification algorithm's robustness has been enhanced to handle edge cases and correctly assess signature validity"

* zlib: Out-of-bounds pointer arithmetic in `inftrees.c` link:https://access.redhat.com/security/cve/CVE-2016-9840[(CVE-2016-9840)]
* krb5: Kerberos RC4-HMAC-MD5 checksum vulnerability link:https://access.redhat.com/security/cve/CVE-2025-3576[(CVE-2025-3576)]

[id="about-release-475_{context}"]
== About release version 4.7.5

*Release date*: 15 July 2025

This release of {product-title-short} 4.7 provides the following bug fixes:

//ROX-29320
* Before this update, incorrect interpretation of Red Hat Enterprise Linux (RHEL) 10 Common Platform Enumeration (CPE) strings caused Scanner V4 to fail distribution checks on RHEL 10 systems. With this update, an updated RHEL CPE major version pattern resolves the issue, and Scanner V4 can now correctly support RHEL 10.

//ROX-29755

> * Before this update, the failure of Sensor to call stream.Recv() caused gRPC flow control to block image reprocessing every 4 hours. With this update, the reprocessing loop includes a timeout for sending messages to Sensors, which resolves the issue and and resumes the image reprocessing as expected.

Reviewer 3: ```suggestion
* Before this update, the failure of Sensor to call stream.Recv() caused gRPC flow control to block image reprocessing every 4 hours. With this update, the reprocessing loop includes a timeout for sending messages to Sensors, which resolves the issue and resumes the image reprocessing as expected.
```

//ROX-28665

> * Before this update, missing network policy configuration for port `9090` on long-running clusters from earlier releases prevented Elastic search from indexing those clusters. With this update, Elastic search integration for long-running cluster metrics resolves the issue, and you can now access search results from these clusters.

Reviewer 3: ```suggestion
* Before this update, missing network policy configuration for port 9090 on long-running clusters from earlier releases prevented Elasticsearch from indexing those clusters. With this update, Elasticsearch integration for long-running cluster metrics resolves the issue, and you can now access search results from these clusters.
```

Reviewer 4: > Afaik this is only internal and does not affect any customers, as they would never deploy a long-running cluster and thus never extract metrics into Elastic Search. I would suggest to skip this paragraph. Pls check me @JoukoVirtanen

Customers do not run long-running clusters, so there is no reason for this to be in the user facing documentation.

Reviewer 5: I am confused, the conversation is resolved, but the text is still there. We suggested to remove that paragraph totally. Did you forget to push maybe?

//ROX-29649

> * Before this update, repeated access to the Singleton instance caused excessive logging of telemetry collection status, resulting in log spam over a 5 day period. With this update, removal of the telemetry collection log from the `InstanceConfig` function resolves the issue and reduces unnecessary log entries.

Reviewer 5: I think, we share too many code-related details that are irrelevant to the users.
```suggestion
* Before this update,  one could observe excessive logging of telemetry collection status, resulting in log spam. With this update, the telemetry collection has been configured to not emit repeated logs continuously, which significantly reduces the log volume.
```

//ROX-29716

> * Before this update, image digest mismatches caused signature verification to fail, even when the signatures were valid. With this update, a check verifies image references against signatures, resolving the issue and allows {ocp} to successfully verify image signatures for secure deployment.

Reviewer 3: ```suggestion
* Before this update, image digest mismatches caused signature verification to fail, even when the signatures were valid. With this update, a check verifies image references against signatures, resolving the issue and allowing {ocp} to successfully verify image signatures for secure deployment.
```
